Library/Homebrew/bottle_checksum.rb

```diff
@@ -0,0 +1,33 @@
+# typed: true
+# frozen_string_literal: true
+
+# A formula's checksum.
+#
+# @api private
+class BottleChecksum
+  extend Forwardable
+
+  attr_reader :hash_type, :hexdigest, :tag, :cellar
+
+  TYPE = :sha256
+
+  def initialize(hexdigest, tag = nil, cellar = nil)
+    @hash_type = TYPE
```

Comment:
Feels weird to have this be a variable that only ever has a constant value

Reply:
I made BottleChecksum a child class of Checksum, this method is not needed anymore. Checksum comparison can directly be done by the parent method.